src/main/java/com/napier/sem/mappers/reports/LanguageReportRowMapper.java

```diff
@@ -0,0 +1,36 @@
+// code by viva
+
+package com.napier.sem.mappers.reports;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.napier.sem.models.reports.LanguageReportRow;
+
+
+public class LanguageReportRowMapper {
+	
+	public static List<LanguageReportRow> GenerateLanguageReportFromResultSet(ResultSet rs) throws SQLException {
+        List<LanguageReportRow> objectList = new ArrayList<LanguageReportRow>();
+
+        while(rs.next()) {
+
+            try
+            {
+            	LanguageReportRow languageReportRow = new LanguageReportRow();
+                languageReportRow.language = rs.getString("Language");
+                languageReportRow.speakers = rs.getDouble("speakers");

```

Comment:
Since speakers needs to be replaced with an integer, you need to convert rs.getDouble("speakers") to an integer like so: (int) Math.round(rs.getDouble("speakers"));